src/v/cluster/controller_backend.cc

```diff
@@ -853,6 +858,23 @@ ss::future<> controller_backend::add_to_shard_table(
       });
 }
 
+ss::future<bool> controller_backend::insert_to_shard_table(
```

Comment:
there is a method inserting ntp to shard table in controller backend already: `add_to_shard_table`

Reply:
There is an issue with using the existing `update` method, I actually had solved a bug due to this. The signature of update takes a raft group id value. Since group id's have no bearing in coproc I figured it would be best to make a new method, ideas?